drivers/mtd/mtd_progmem.c

```diff
@@ -131,7 +131,7 @@ static struct progmem_dev_s g_progmem =
  *
  ****************************************************************************/
 
-static int32_t progmem_log2(uint32_t blocksize)
+static int32_t progmem_log2(size_t blocksize)
```

Comment:
i guess you should use unsigned long to match https://github.com/apache/incubator-nuttx/blob/23495a7fa26368cde373e7091631de2a1926012b/include/nuttx/mtd/mtd.h#L191

Reply:
@yamt: I don't think so. The various _ioctl_ function pointers in _include/nuttx/fs/ioctl.h_ and elsewhere have _arg_ as unsigned long because that is supposed to be a generic argument that can represent other things than just sizes of memory areas.